### PR TITLE
Virt operator: improve test time

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
This series of commits gradually remove complexity of pool tests to the point where we reduce the runtime from ~22 seconds down to ~200ms. On top of this we remove some unnecessary code and fix few issues.

### Special notes for your reviewer
These changes follow the logic introduced here https://github.com/kubevirt/kubevirt/pull/12468

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
based on https://github.com/kubevirt/kubevirt/pull/12537

**Comment by xpivarc** [comment]
@fossedihelm Can we split this PR to 2?

**Comment by fossedihelm** [comment]
> @fossedihelm Can we split this PR to 2?

@xpivarc I will be happy to do it, but I cannot find how we can do it.
If the problem is the commit number I can do smth like:
- Remove all the fakeControllerSource
- Use controller stores
- Remove config

Besides the number of commits, I cannot find any problematic things in this PR. Or am I wrong? Thank you!

File: pkg/virt-operator/kubevirt_test.go [hunk 1]
```diff
@@ -1589,16 +1587,16 @@ func (k *KubeVirtTestData) addPodDisruptionBudgets(config *util.KubeVirtDeployme
...
+	k.informers.Namespace.GetStore().Update(&k8sv1.Namespace{
 		TypeMeta: metav1.TypeMeta{
 			Kind: "Namespace",
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Name: NAMESPACE,
 		},
 	})
-	k.mockQueue.Wait()
+	key := kubecontroller.NamespacedKey("default", NAMESPACE)
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-operator/kubevirt_test.go:1563]: It is not correct :) Nice catch

**Review by 0xFelix — APPROVED** [review]
/lgtm 

Thanks!

**Pull request merged**